### PR TITLE
Add cql-based cassandra backend

### PR DESCRIPTION
**Pull request by dbenamydd**
This was originally reviewed in #1; I just created a new PR with `datadog` as the destination so `master` can continue to track upstream in case they wake up.

### CODE REVIEW

**Comment by dbenamydd** [comment]
Hey @sethrosenblum! Do you feel like reviewing some follow up work before I merge this? The first few commits are the same as what you already reviewed. The new ones start at `Move assertion to base class	04b4e59`.

The key thing is that the previous implementation would create a new connection to the cluster for every request (and prepare the statements each time). This pulls apart the backend into 2 classes- 1 that manages the interaction with cassandra, and 1 that talks beaker and keeps a persistent instance of the 1st around.

And while I was at it I made a few other improvements.

File: setup.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 version = '0.2.0'
```
Comment (LotharSee) [inline setup.py]: Don't forget to update the version and tag once merged.

File: beaker_extensions/cassandra_cql.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                 **params):
+        if not keyspace:
+            raise MissingCacheParameter("keyspace is required")
+        if re.search(r'[^0-9a-zA-Z_]', keyspace):
+            raise ValueError(
+                "keyspace can only have alphanumeric chars and underscore"
+            )
+        self.__keyspace_cql_safe = keyspace
+        table = column_family or 'beaker'
+        if re.search(r'[^0-9a-zA-Z_]', table):
```
Comment (LotharSee) [inline beaker_extensions/cassandra_cql.py]: This regex can be simplified as `r'\w'`. You could also compile it once, but that's really not important.

**Comment by LotharSee** [comment]
What is really important here is the error handling. We got some issues with the Redis backend, not catching everything. Combined with Pylons, it was crashing before the initialization of our custom loggers, making it silent and hard to catch.

So let's double/triple check we catch any kind of exceptions, deal with any state for the Cassandra cluster and answer accordingly.

File: beaker_extensions/cassandra_cql.py (hunk 2 above)
Comment (dbenamydd) [inline beaker_extensions/cassandra_cql.py]: Actually `r'\W'` :-) Done!

File: setup.py (hunk 1 above)
Comment (dbenamydd) [inline setup.py]: Thanks. Will do.

**Comment by dbenamydd** [comment]
Moving convo to other repo.

**Comment by dbenamydd** [comment]
It sounds like everyone's happ with this so I'll merge to unblock further work. If I misunderstood, just let me know; this won't be going live immediately.

**Pull request merged**